src/index.js

```diff
@@ -157,12 +177,15 @@ class DropdownTreeSelect extends Component {
     })
   }
 
-  onTagRemove = (id, isKeyboardEvent) => {
+  onTagRemove = id => {
     const { tags: prevTags } = this.state
     this.onCheckboxChange(id, false, tags => {
-      if (!isKeyboardEvent) return
-
-      keyboardNavigation.getNextFocusAfterTagDelete(id, prevTags, tags, this.searchInput).focus()
+      const nextFocus = keyboardNavigation.getNextFocusAfterTagDelete(id, prevTags, tags, this.searchInput)
+      if (nextFocus) {
+        nextFocus.focus()
+      } else {
+        this.onBlur()
```

Comment:
This may not be an onBlur as the user would still be in the input, wouldn't they?

Reply:
i can not see a case where the component is on focus. if the focused element is removed, and there is nothing else you can focus on, doesn't that mean the component loses focus? what am i missing?